frontend/apps/vara-man/src/feature/single-game/components/timer/index.tsx

```diff
@@ -2,40 +2,45 @@ import { useEffect, useState } from 'react';
 import { useAtom } from 'jotai';
 import { GAME_OVER } from '@/feature/game/consts';
 
-type Props = {
-  gameRestarted: boolean;
-};
+const totalTimeGame = 10 * 60 * 1000;
 
-export const GameTimer = ({ gameRestarted }: Props) => {
-  const [, setGameOver] = useAtom(GAME_OVER);
-  const [timeLeft, setTimeLeft] = useState('');
-  const [startTime, setStartTime] = useState<number | null>(null);
+export const GameTimer = () => {
+  const [gameOver, setGameOver] = useAtom(GAME_OVER);
+  const [remainingTime, setRemainingTime] = useState(totalTimeGame);
 
+  const [startTime, setStartTime] = useState<number | null>(null);
   useEffect(() => {
+    if (gameOver) return;
+
     const start = Date.now();
     setStartTime(start);
 
-    const totalGameTime = 10 * 60 * 1000;
-
     const updateTimer = () => {
       const now = Date.now();
       const elapsedTime = now - (startTime || now);
-      const remainingTime = totalGameTime - elapsedTime;
+      const newRemainingTime = remainingTime - elapsedTime;
 
-      if (remainingTime <= 0) {
-        setTimeLeft('00:00');
+      if (newRemainingTime > 0) {
+        setRemainingTime(newRemainingTime);
+      } else if (newRemainingTime <= 0) {
         setGameOver(true);
-      } else {
-        const minutes = Math.floor((remainingTime / (1000 * 60)) % 60);
-        const seconds = Math.floor((remainingTime / 1000) % 60);
-        setTimeLeft(`${minutes < 10 ? '0' : ''}${minutes}:${seconds < 10 ? '0' : ''}${seconds}`);
+        setRemainingTime(totalTimeGame);
       }
     };
 
     const timerId = setInterval(updateTimer, 1000);
 
     return () => clearInterval(timerId);
-  }, [setGameOver, startTime, gameRestarted]);
 
-  return <>{timeLeft}</>;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [startTime, gameOver, remainingTime]);
+
+  const formatterTime = () => {
```

Comment:
It would be great to take this to `utils`.

Seems like same function is used in these files:
`vara-man\src\feature\tournament-game\components\game-players\index.tsx:75`
`vara-man\src\feature\tournament-game\components\game-players\index.tsx:147`
`vara-man\src\feature\tournament-game\components\timer\index.tsx:38`

Reply:
And also good name for it is `getFormattedTime`. Like in `battleship\src\features\game\utils.ts:35`